### PR TITLE
Added flat_array structtag to V2 APIs to fix query param encoding

### PR DESCRIPTION
**Pull request by prathmesh-stripe**
### Why?
See V2 form encoding problem in Go reported in https://github.com/stripe/stripe-go/issues/2181

### What?
- In V2(url query params) we want the arrays to be encoded flat as `foo=bar&foo=baz`. 
- In V1, arrays are still index-encoded as `foo[0]=bar&foo[1]=baz`.

### Test Plan
- [x] All changes in PR are covered by tests
- [ ] Failures and edge cases tested
- [x] Manually tested any SDK changes

### See Also
https://github.com/stripe/stripe-go/issues/2181

## Changelog
* Fixes URL serialization for array query parameters in V2 APIs.

### CODE REVIEW

File: form/form.go [hunk 1]
```diff
@@ -408,6 +418,7 @@ func makeStructEncoder(t reflect.Type) *structEncoder {
 	// `form`
 	se := &structEncoder{}
 
+	// validation step
```
Comment (mbroshi-stripe) [inline form/form.go]: Couple of things:
1. Did you remove the extra validation we added for `flat_array` on purpose?
2. This comment belongs above line 444. Can you move it there, and expand on this comment a little? Maybe a sentence like `// validate that fields are of expected types and have expected annotations`

Comment (prathmesh-stripe) [reply form/form.go]: 1. Yes, it was also running on v1 APIs which is not we want. 
2. Sure

Comment (prathmesh-stripe) [reply form/form.go]: 1. Added extra validation back. The test failure was due to option nil check.

File: v2/core/eventdestination/client_test.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ func TestEventDestinationGet(t *testing.T) {
 		assert.NoError(t, err)
 		return data
 	}, func(t *testing.T, r *http.Request) {
-		assert.Equal(t, r.URL.Query().Get("include[0]"), "webhook_endpoint.url")
+		assert.Equal(t, r.URL.Query().Get("include"), "webhook_endpoint.url")
```
Comment (prathmesh-stripe) [inline v2/core/eventdestination/client_test.go:93]: This was changed because of a failing test.

**Comment by prathmesh-stripe** [comment]
> This looks like it has some unrelated generated changes pulled in; mostly descriptions and tests, but is that intentional?
@jar-stripe 

Those were unintentional but they were generated using the last GA release spec(v2028). They were comment changes so didn't look too closely at them.
